packages/cadl-python/src/emitter.ts

```diff
@@ -742,27 +798,87 @@ function emitBasicOperation(
         }
     }
     const name = camelToSnakeCase(getLibraryName(context, operation));
-    return [
-        {
-            name: name,
-            description: getDocStr(context, operation),
-            summary: getSummary(context.program, operation),
-            url: httpOperation.path,
-            method: httpOperation.verb.toUpperCase(),
-            parameters: parameters,
-            bodyParameter: bodyParameter,
-            responses: responses,
-            exceptions: exceptions,
-            groupName: operationGroupName,
-            addedOn: getAddedOnVersion(context, operation),
-            discriminator: "basic",
-            isOverload: false,
-            overloads: [],
-            apiVersions: [getAddedOnVersion(context, operation)],
-            wantTracing: true,
-            exposeStreamKeyword: true,
-        },
-    ];
+
+    // handle native overloads
+    let overloads: Record<string, any>[] = [];
+    const overload_operations = getOverloads(context.program, operation);
+    if (overload_operations && bodyParameter) {
+        for (const overload_operation of overload_operations) {
+            overloads = overloads.concat(emitBasicOperation(context, overload_operation, operationGroupName, true));
+        }
+        let binaryOverload = undefined;
```

Comment:
we can't assume every overload situation is 1 binary overload and 1 model overload. One example is this cadl

```cadl
op func(body: str | List[str]): void

@overload(func)
op funcStr(body: str): void

@overload(func)
op funcListStr(body: str[]): void
```
```

Reply:
currently, we only generate free IO overload for `bytes or model` and free JSON overload for `model`. If no model and no bytes, we don't generate the free overload. Is that expected?